external-crates/move/crates/move-bytecode-verifier-meter/src/lib.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use move_binary_format::errors::PartialVMResult;
+use std::ops::Mul;
+
+pub mod bound;
+pub mod dummy;
+
+/// Scope of metering
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum Scope {
+    // Metering is for transaction level
+    Transaction,
+    // Metering is for package level
+    Package,
+    // Metering is for module level
+    Module,
+    // Metering is for function level
+    Function,
+}
+
+/// Trait for a metering verification.
+pub trait Meter {
+    /// Indicates the begin of a new scope.
+    fn enter_scope(&mut self, name: &str, scope: Scope);
+
+    /// Transfer the amount of metering from once scope to the next. If the current scope has
+    /// metered N units, the target scope will be charged with N*factor.
+    fn transfer(&mut self, from: Scope, to: Scope, factor: f32) -> PartialVMResult<()>;
+
+    /// Add the number of units to the meter, returns an error if a limit is hit.
+    fn add(&mut self, scope: Scope, units: u128) -> PartialVMResult<()>;
+
+    /// Adds the number of items.
+    fn add_items(
+        &mut self,
+        scope: Scope,
+        units_per_item: u128,
+        items: usize,
+    ) -> PartialVMResult<()> {
+        if items == 0 {
```

Comment:
I find these sort of micro-optimizations peculiar but fine by me

Reply:
This is just a straight copy of the existing `Meter` trait, I wouldn't be opposed to revisiting it, but didn't want to do that in this PR to not muddy the history too much.